core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -208,7 +209,7 @@ private class EncryptedBlockData(
     conf: SparkConf,
     key: Array[Byte]) extends BlockData {
 
-  override def toInputStream(): InputStream = Channels.newInputStream(open())
+  override def toInputStream(): InputStream = new NioBufferedFileInputStream(file)
```

Comment:
will `NioBufferedFileInputStream` release its memory automatically?

Reply:
You meaning the memory buffer? The `NioBufferedFileInputStream` has `close` method, you can see follow:
org.apache.spark.io.NioBufferedFileInputStream.java
```java
@Override
  public synchronized void close() throws IOException {
    fileChannel.close();
    StorageUtils.dispose(byteBuffer);
  }

  @Override
  protected void finalize() throws IOException {
    close();
  }
```